ni_measurementlink_service/measurement/service.py

```diff
@@ -513,6 +532,9 @@ def _make_annotations_dict(
         if type_specialization == TypeSpecialization.Pin:
             if instrument_type != "" or instrument_type is not None:
                 annotations["ni/pin.instrument_type"] = instrument_type
+        if type_specialization == TypeSpecialization.IOResource:
```

Comment:
Note: We eventually want to add an io_type specialization and some annotations for attributes. See this [PR comment thread](https://dev.azure.com/ni/DevCentral/_git/ASW/pullRequest/681888#1711575373) for some context. That doesn't have to be done as part of this PR, but I thought I would bring it to your attention.

Reply:
I'd prefer to deal with this in a separate PR. I'll work with @DelpireNI to figure out how to best capture the work.

Reply:
AB#2711389